### DISCUSSION TITLE
Dual averaging for tempered SMC target tempering

### DISCUSSION THREAD

**Discussion by AdrienCorenflos** [blackjax-devs/blackjax#114] [New features]
It feels to me like it would be an easy improvement to wrap DA in order to provide an ESS solver. WDYT?

**Comment by rlouf** [comment]
Go for it!

**Comment by junpenglao** [comment]
🤖 *This reply was drafted with Claude and reviewed by me (@junpenglao).*

This is effectively covered by `blackjax.adaptive_tempered_smc`, which solves for the next temperature to hit a target ESS at each step (in place of a hand-tuned tempering schedule). Closing as resolved — reopen if you meant DA specifically for a different quantity.
